tests/IResearch/IResearchQueryJoin-test.cpp

```diff
@@ -1496,13 +1496,8 @@ TEST_F(IResearchQueryJoinTest, test) {
         "collection_1 FILTER x.seq == d.seq && x.name == d.name SORT "
         "customscorer(d, x.seq) DESC RETURN d";
 
-    EXPECT_TRUE(arangodb::tests::assertRules(vocbase, query,
-                                             {
-                                                 arangodb::aql::OptimizerRule::handleArangoSearchViewsRule,
-                                             }));
-
     auto queryResult = arangodb::tests::executeQuery(vocbase, query);
-    ASSERT_TRUE(queryResult.result.is(TRI_ERROR_BAD_PARAMETER));
+    ASSERT_TRUE(queryResult.result.is(TRI_ERROR_INTERNAL));
```

Comment:
This is breaking change. We have to understand why `handleArangoSearchViewsRule` is not applied anymore and avoid changing error code.